diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
index d9ae4fe5b..e60f35780 100644
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -225,6 +225,13 @@ config("runtime_library") {
   cflags = []
   cflags_cc = []
 
+  cflags += [
+    "-Wno-unknown-argument", "-Wno-unknown-warning-option",
+    "-D_SILENCE_ALL_CXX20_DEPRECATION_WARNINGS"
+  ]
+
+  ldflags = [ "/LIBPATH:" + getenv("VCINSTALLDIR") + "\\Tools\\MSVC\\" + getenv("VCToolsVersion") + "\\lib\\x64" ]
+
   # Defines that set up the CRT.
   defines = [
     "__STD_C",
diff --git a/toolchain/win/tool_wrapper.py b/toolchain/win/tool_wrapper.py
index 47bbfe2a0..3bec9006d 100644
--- a/toolchain/win/tool_wrapper.py
+++ b/toolchain/win/tool_wrapper.py
@@ -140,7 +140,7 @@ class WinTool(object):
     # Read output one line at a time as it shows up to avoid OOM failures when
     # GBs of output is produced.
     for line in link.stdout:
-      line = line.decode('utf8')
+      line = line.decode('cp1252')
       if (not line.startswith('   Creating library ')
           and not line.startswith('Generating code')
           and not line.startswith('Finished generating code')):
